### ISSUE TITLE
react/destructuring-assignment oneliner

### ISSUE THREAD

**Issue by Fandekasp** [yannickcr/eslint-plugin-react#1850]
Simple and clean oneliner:
```es6
  handleClick = () => this.props.firebase.logout();
```

Right now, I have to write
```es6
  handleClick = () => {
    const {firebase} = this.props;
    firebase.logout();
  };
```

This destructuration is useless in this case. Please remove warnings for oneliner methods where no params are given to the method.

**Comment by ljharb** [comment]
It's definitely not useless; in this case, it doesn't cover the most dangerous problem:
```jsx
handleClick = () => this.props.foo();
// passes `this.props` as the `this` value, which exposes it to code that could mutate it.
```

Separately, you never want to put arrow functions in class fields, so I wouldn't recommend that being a one-liner anyways :-)

**Comment by ljharb** [comment]
The props object *should not* be mutated, but is in no way immutable.

As for arrow functions in class fields, you always want instance methods with a constructor-binding (doing a constructor-bind along with an arrow function is redundant) for performance, clarity, and testability.

**Comment by Fandekasp** [comment]
```
  handleClick () {
    const {firebase} = this.props;
    firebase.logout();
  }
```
it is, then. Looking forward to the rule, please share it in this issue if you don't forget about it by then, thanks :)

**Comment by jeffvandyke** [comment]
@ljharb ...until decorators are also stage 4, then an `@autobind` on the handler is much less jumping around than binding in constructors.

**Comment by ljharb** [comment]
@jeffvandyke sure. With class fields, you can also do:
```jsx
foo = this.foo.bind(this);
foo() { }
```
